### PR TITLE
Proguard

### PR DESCRIPTION
**Pull request by LeafHacker**
Apply @leijurv's proguard config to optimise the Jar size. Also add a manifest.xml for the windows exe so we can specify that we don't need admin perms.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import proguard.gradle.ProGuardTask
```
Comment (leijurv) [inline build.gradle]: not a blocker but i think this goes below the copyright header ideally?

File: build.gradle [hunk 2]
```diff
@@ -54,18 +67,51 @@ jar {
...
+
+    verbose()
+    dontwarn()
+
+    keepattributes 'Signature'
+    keepattributes '*Annotation*'
+    optimizationpasses 9
+    dontusemixedcaseclassnames()
+    
+    keepdirectories 'icons'
```
Comment (leijurv) [inline build.gradle]: nice!

File: exe/manifest.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~ along with Impact Installer.  If not, see <https://www.gnu.org/licenses/>.
+  -->
+
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+    <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+        <security>
+            <requestedPrivileges>
+                <requestedExecutionLevel level="asInvoker" uiAccess="false"/>
```
Comment (leijurv) [inline exe/manifest.xml]: have u verified that this is actually able to modify %appdata% and install, with these reduced permissions?

File: build.gradle [hunk 4]
```diff
@@ -54,18 +67,51 @@ jar {
...
 
 shadowJar {
-    classifier = ''
+    classifier = 'unoptimised'
+}
+
+task('optimize', type: ProGuardTask, dependsOn: shadowJar) {
+    injars tasks.shadowJar.outputs.files
+    outjars "${buildDir}/libs/${project.tasks.jar.archiveName}"
+    libraryjars "${System.getProperty('java.home')}/lib/rt.jar"
```
Comment (leijurv) [inline build.gradle]: i don't know if this is an issue its probably fine but i have no idea

does this need to be java 8? or does it work on later ones

File: exe/manifest.xml (hunk 3 above)
Comment (LeafHacker) [reply exe/manifest.xml]: AppData is always user-editable. It is within the user's home directory and is where all their caching and config is done

File: build.gradle (hunk 4 above)
Comment (LeafHacker) [reply build.gradle]: IDK, everyone seems to use rt.jar (including you in the .pro you sent). I assume it works for all versions

File: exe/manifest.xml (hunk 3 above)
Comment (leijurv) [reply exe/manifest.xml]: cool

File: build.gradle (hunk 1 above)
Comment (LeafHacker) [reply build.gradle]: I'd agree, however that's where intellij put it :/ can move it i'd just worry intellijj would be weird next time it imports something. Alternatively we could use a canonical reference

File: build.gradle (hunk 4 above)
Comment (leijurv) [reply build.gradle]: coolio

**Pull request merged**